### ISSUE TITLE
[axe-core-warning] Ensures every form element has a label

### ISSUE THREAD

**Issue by nramamurth** [cerner/terra-core#2570]
# Bug Report
Accessibility is violated on a few TextArea test pages and had a consumer tell me that they ran into this issue while running their _wdio_ tests as well.

<details>
 <summary>WDIO Axe Error Report</summary>

```
"description": "Ensures every form element has a label",
  "help": "Form elements must have labels",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/label?application=axeAPI",
  "id": "label",
  "impact": "critical",
  "nodes": [
    {
      "all": [],
      "any": [
        {
          "data": null,
          "id": "aria-label",
          "impact": "serious",
          "message": "aria-label attribute does not exist or is empty",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "aria-labelledby",
          "impact": "serious",
          "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "implicit-label",
          "impact": "critical",
          "message": "Form element does not have an implicit (wrapped) <label>",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "explicit-label",
          "impact": "critical",
          "message": "Form element does not have an explicit <label>",
          "relatedNodes": []
        },
        {
          "data": null,
          "id": "non-empty-title",
          "impact": "serious",
          "message": "Element has no title attribute or the title attribute is empty",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  aria-label attribute does not exist or is empty\n  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty\n  Form element does not have an implicit (wrapped) <label>\n  Form element does not have an explicit <label>\n  Element has no title attribute or the title attribute is empty",
      "html": "<textarea id=\"auto-resizable\" cols=\"2\" rows=\"2\" class=\"Textarea-module__textarea___2LAQp\" style=\"min-height: 48px; caret-color: transparent;\">Default Value</textarea>",
      "impact": "critical",
      "none": [],
      "target": [
        "#auto-resizable"
      ]
    }
  ],
  "tags": [
    "cat.forms",
    "wcag2a",
    "wcag332",
    "wcag131",
    "section508",
    "section508.22.n"
  ]

```
</details>

## Steps to Reproduce

1. Go to https://engineering.cerner.com/terra-core/tests/terra-form-textarea/form-textarea/auto-resizable-textarea
2. Run the axe test.

### Additional Information
It looks like the rule is being ignored in the _wdio_ tests [1] for TextArea. Is this for a reason?

[1] https://github.com/cerner/terra-core/blob/21630dbf90a05bb93567aeba3a687843ad142246/packages/terra-form-textarea/tests/wdio/form-textarea-spec.js#L1-L3

**Comment by bjankord** [comment]
Talked about this in our sync today. We want to move forward with adding a label prop to Input, Textarea, and Select.

**Comment by RA053031** [comment]
Hi,
I am using form-input, and facing the same issue. Can this be prioritised and fixed at the earliest?

**Comment by pranav300** [comment]
@bjankord I think we can remove `form-select` from the issue as each component in it uses Frame.jsx which has the prop `ariaLabel`. What do you say?

**Comment by bjankord** [comment]
@pranav300 sounds good
